spec/controllers/admin/customers_controller_spec.rb

```diff
@@ -171,4 +171,158 @@ def create_customer(enterprise)
+        end
+
+        it 'invokes after destroy callbacks' do
+          allow(controller).to receive(:location_after_destroy) { '/fix_location_after_destroy' }
+          allow(controller).to receive(:invoke_callbacks).with(:destroy, :before)
+
+          expect(controller).to receive(:invoke_callbacks).with(:destroy, :after)
+          spree_delete :destroy, id: customer, format: :html
+        end
+      end
+
+      context 'and the destroy fails' do
+        before do
+          allow(Customer).to receive(:find).with(customer.id.to_s).and_return(customer)
+          allow(customer).to receive(:destroy).and_return(false)
+        end
+
+        it 'invokes before destroy callbacks' do
+          allow(controller).to receive(:location_after_destroy) { '/fix_location_after_destroy' }
+          allow(controller).to receive(:invoke_callbacks).with(:destroy, :fails)
+
+          expect(controller).to receive(:invoke_callbacks).with(:destroy, :before)
+          spree_delete :destroy, id: customer, format: :html
+        end
+
+        it 'raises due to missing #location_after_destroy' do
+          expect {
+            spree_delete :destroy, id: customer, format: :html
+          }.to raise_error(NameError, /location_after_destroy/)
+        end
+
+        it 'invokes failure destroy callbacks' do
+          allow(controller).to receive(:location_after_destroy) { '/fix_location_after_destroy' }
+          allow(controller).to receive(:invoke_callbacks).with(:destroy, :before)
+
+          expect(controller).to receive(:invoke_callbacks).with(:destroy, :fails)
+          spree_delete :destroy, id: customer, format: :html
+        end
+      end
+    end
+
+    context 'when rendering js' do
+      context 'and the destroy succeeds' do
+        it 'invokes before destroy callbacks' do
+          allow(controller).to receive(:invoke_callbacks).with(:destroy, :after)
+
+          expect(controller).to receive(:invoke_callbacks).with(:destroy, :before)
+          spree_delete :destroy, id: customer, format: :js
+        end
+
+        it 'renders the spree/admin/shared/destroy partial' do
+          spree_delete :destroy, id: customer, format: :js
+          expect(response).to render_template('spree/admin/shared/_destroy')
+        end
+
+        it 'invokes after destroy callbacks' do
+          allow(controller).to receive(:invoke_callbacks).with(:destroy, :before)
+
+          expect(controller).to receive(:invoke_callbacks).with(:destroy, :after)
+          spree_delete :destroy, id: customer, format: :js
+        end
+      end
+
+      context 'and the destroy fails' do
+        before do
+          allow(Customer).to receive(:find).with(customer.id.to_s).and_return(customer)
+          allow(customer).to receive(:destroy).and_return(false)
+        end
+
+        it 'raises due to missing template' do
+          expect {
+            spree_delete :destroy, id: customer, format: :js
+          }.to raise_error(ActionView::MissingTemplate)
```

Comment:
It turns out that we never used checked the scenario the destroy didn't work when requesting it with JS. We need to cover it somehow.